features/telemetry/protocols/flows/src/main/java/org/opennms/netmgt/telemetry/protocols/flows/AbstractFlowAdapter.java

```diff
@@ -108,7 +112,9 @@ public void handleMessageLog(TelemetryMessageLog messageLog) {
 
         try {
             LOG.debug("Persisting {} packets, {} flows.", flowPackets.size(), flows.size());
-            final FlowSource source = new FlowSource(messageLog.getLocation(), messageLog.getSourceAddress());
+            final FlowSource source = new FlowSource(messageLog.getLocation(),
+                    messageLog.getSourceAddress(),
+                    Strings.isNullOrEmpty(metaDataNodeLookup) ? null : new ContextKey(metaDataNodeLookup));
```

Comment:
We can store the `ContextKey` object in a local variable rather than rebuilding it for every log processed.

Reply:
done